crates/vm2-interface/src/state_interface.rs

```diff
@@ -1,93 +1,148 @@
 use primitive_types::{H160, U256};
 
+/// Public interface of the VM state. Encompasses both read and write methods.
 pub trait StateInterface {
+    /// Reads a register with the specified zero-based index. Returns a value together with a pointer flag.
     fn read_register(&self, register: u8) -> (U256, bool);
+    /// Sets a register with the specified zero-based index
     fn set_register(&mut self, register: u8, value: U256, is_pointer: bool);
 
+    /// Returns a mutable handle to the current call frame.
     fn current_frame(&mut self) -> impl CallframeInterface + '_;
+    /// Returns the total number of call frames.
     fn number_of_callframes(&self) -> usize;
+    /// Returns a mutable handle to a call frame with the specified index, where
     /// zero is the current frame, one is the frame before that etc.
     fn callframe(&mut self, n: usize) -> impl CallframeInterface + '_;
 
-    fn read_heap_byte(&self, heap: HeapId, index: u32) -> u8;
-    /// Reads an entire `U256` word in the big-endian order from the specified heap page / `index`
+    /// Reads a single byte from the specified heap page at the specified 0-based offset.
+    fn read_heap_byte(&self, heap: HeapId, offset: u32) -> u8;
+    /// Reads an entire `U256` word in the big-endian order from the specified heap page / `offset`
```

Comment:
I try to always write just "heap" rather than "memory page" because the latter's established meaning related to mapping physical memory to virtual memory.

```suggestion
    /// Reads an entire `U256` word in the big-endian order from the specified heap at `offset`
```